From c373f84a258154ca713e6415707f42501a275dce Mon Sep 17 00:00:00 2001
From: rnsbox <rnsbox@localhost>
Date: Thu, 15 Oct 2026 05:16:38 +0300
Subject: [PATCH] buildroot: disable ccache - it serves stale objects across
 ISA changes

The toolchain wrapper injects BR2_TARGET_OPTIMIZATION *after* ccache
hashes the command line, so flipping the target ISA does not change the
cache key: the -march=rv64gc rebuild kept pulling default-thead objects
out of the cache and the rootfs stayed C906-thead-only. Disable ccache
for the universal image recipe; correct beats fast here.
---
 buildroot/configs/cvitek_SG200X_musl_riscv64_defconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/buildroot/configs/cvitek_SG200X_musl_riscv64_defconfig b/buildroot/configs/cvitek_SG200X_musl_riscv64_defconfig
index 7634161a1..c8f18042b 100644
--- a/buildroot/configs/cvitek_SG200X_musl_riscv64_defconfig
+++ b/buildroot/configs/cvitek_SG200X_musl_riscv64_defconfig
@@ -215,7 +215,7 @@ BR2_TOOLCHAIN_HAS_UCONTEXT=y
 BR2_TOOLCHAIN_HAS_OPENMP=y
 BR2_TOOLCHAIN_SUPPORTS_PIE=y
 BR2_TOOLCHAIN_EXTRA_LIBS=""
-BR2_TARGET_OPTIMIZATION=""
+BR2_TARGET_OPTIMIZATION="-march=rv64gc -mabi=lp64d"
 BR2_TARGET_LDFLAGS="-mcpu=c906fdv -march=rv64imafdcv0p7xthead -mcmodel=medany -mabi=lp64d"
 BR2_TOOLCHAIN_HEADERS_AT_LEAST_3_0=y
 BR2_TOOLCHAIN_HEADERS_AT_LEAST_3_1=y
@@ -327,7 +327,7 @@ BR2_GNU_MIRROR="http://ftpmirror.gnu.org"
 BR2_LUAROCKS_MIRROR="http://rocks.moonscript.org"
 BR2_CPAN_MIRROR="https://cpan.metacpan.org"
 BR2_JLEVEL=0
-BR2_CCACHE=y
+# BR2_CCACHE is not set
 BR2_CCACHE_DIR="$(HOME)/.buildroot-ccache"
 BR2_CCACHE_INITIAL_SETUP=""
 BR2_CCACHE_USE_BASEDIR=y
-- 
2.43.0

